### PR TITLE
remove GAed feature gate:SCTPSupport

### PR DESCRIPTION
**Pull request by wangyysde**
Signed-off-by: wangyysde <yuying.wang@daocloud.io>

#### What type of PR is this?
/kind feature
/kind cleanup

#### What this PR does / why we need it:
#95566 has moved SCTP to GA and locks the feature-gate on.Additional, the relevant code for feature-gate checks has been removed.  This PR removes the feature-gate.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
-[KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/0015-20180614-SCTP-support.md
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -815,7 +815,6 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	CSIVolumeFSGroupPolicy:                         {Default: true, PreRelease: featuregate.Beta},
 	RuntimeClass:                                   {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.23
 	NodeLease:                                      {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```
Comment (pacoxu) [inline pkg/features/kube_features.go:817]: Lines 283-289 need to be removed as well.
/ok-to-test

/cc danwinship janosi
per #95566

**Comment by wangyysde** [comment]
@pacoxu  I have removed them. Thanks!

**Comment by pacoxu** [comment]
/priority important-soon
/lgtm
/assign @thockin 
as it is network feature.

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**